x-pack/plugins/security_solution/server/endpoint/services/agent/clients/sentinel_one/sentinel_one_agent_status_client.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getPendingActionsSummary } from '../../..';
+import type { RawSentinelOneInfo, SentinelOneSearchResponse } from './types';
+import { catchAndWrapError } from '../../../../utils';
+import type { AgentStatusRecords } from '../../../../../../common/endpoint/types';
+import { HostStatus } from '../../../../../../common/endpoint/types';
+import type { ResponseActionAgentType } from '../../../../../../common/endpoint/service/response_actions/constants';
+import { AgentStatusClient } from '../lib/base_agent_status_client';
+import { AgentStatusClientError } from '../errors';
+
+const SENTINEL_ONE_AGENT_INDEX = `logs-sentinel_one.agent-default`;
+
+enum SENTINEL_ONE_NETWORK_STATUS {
+  CONNECTING = 'connecting',
+  CONNECTED = 'connected',
+  DISCONNECTING = 'disconnecting',
+  DISCONNECTED = 'disconnected',
+}
+
+export class SentinelOneAgentStatusClient extends AgentStatusClient {
+  protected readonly agentType: ResponseActionAgentType = 'sentinel_one';
+
+  async getAgentStatuses(agentIds: string[]): Promise<AgentStatusRecords> {
+    const esClient = this.options.esClient;
+    const metadataService = this.options.endpointService.getEndpointMetadataService();
+
+    const query = {
+      bool: {
+        must: [
+          {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    'sentinel_one.agent.uuid': agentIds,
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    };
+
+    try {
+      const response = (await esClient
+        .search(
+          {
+            index: SENTINEL_ONE_AGENT_INDEX,
+            from: 0,
+            size: 10000,
+            query,
+            collapse: {
+              field: 'sentinel_one.agent.uuid',
+              inner_hits: {
+                name: 'most_recent',
+                size: 1,
+                sort: [
+                  {
+                    '@timestamp': {
```

Comment:
Should this be `sentinel_one.agent.last_active_date` here?

`@timestamp` is probably the ingest timestamp and because these are being infested on an interval, they would all received that same date.

Reply:
done 6d789d0d39feb1adc3c3a6514c2f03ff86de9763

Reply:
done a21b9bd763b19fec61da1c3cb3a5430f076aa0ae